controllers/BookmarksController.js

```diff
@@ -0,0 +1,121 @@
+
+import Models from '../models';
+import StatusResponse from '../helpers/StatusResponse';
+
+const { bookmarks, articles } = Models;
+/** @description usersController class
+ * @return {object} the response object
+ * @public
+ */
+class BookmarksController {
+  /** @description create a bookmark
+   * @param {string} req is the request parameter
+   * @param {string} res is the response parameter
+   * @return {object} the response object
+   * @public
+   */
+  static async create(req, res) {
+    const { userId } = res.locals.user;
+    const { articleId } = req.params;
+    let { title } = req.body;
+
+    if (!title) {
+      title = `untitled ${articleId}`;
+    }
+
+    try {
+      const bookmark = await bookmarks.create({
+        title,
+        userId,
+        articleId
+      });
+
+      if (!bookmark) {
+        return StatusResponse.badRequest(res, { message: 'error posting bookmark' });
+      }
+
+      return StatusResponse.success(res, { message: 'hurray! bookmark was added successfully' });
+    } catch (error) {
+      return StatusResponse.internalServerError(error);
+    }
+  }
+
+  /** @description get one bookmark
+   * @param {object} req is the request parameter
+   * @param {object} res is the response parameter
+   * @return {object} the response object
+   * @public
+   */
+  static async search(req, res) {
+    const { title } = req.body;
+    const { userId } = res.locals.user;
+
+    if (!title) {
+      return StatusResponse.badRequest(res, { message: 'please insert a title' });
+    }
+
+    const bookmark = await bookmarks.findAll({
+      where: {
+        title,
+        userId,
+      },
+      include: [{
+        model: articles,
+        as: 'article',
+        attributes: { exclude: ['createdAt', 'updatedAt', 'slug'] }
+      }],
+    });
+    if (!bookmark) {
+      return StatusResponse.notfound(res, { message: 'no bookmark found' });
+    }
+    return StatusResponse.success(res, bookmark);
+  }
+
+  /** @description get all bookmarks
+   * @param {object} req is the request parameter
+   * @param {object} res is the response parameter
+   * @return {object} the response object
+   * @public
+   */
+  static async getAll(req, res) {
+    const { userId } = res.locals.user;
+
+    const bookmark = await bookmarks.findAll({
+      where: {
+        userId
+      },
+      include: [{
+        model: articles,
+        as: 'article',
+        attributes: { exclude: ['createdAt', 'updatedAt', 'slug'] }
```

Comment:
What made you choose to exclude these columns?

Reply:
on the call yesterday you suggested I exclude them, so I don't return all the columns on the article table.. that it will be unnecessary to do that, especially returning the createdAt and updatedAt @Chidi3b3r3

Reply:
If you were to build the client interface for this feature, what fields would you need?
I said on the call to include only the needed fields because building the UI doesn't need most of the fields. My thinking is `article title, article id, article slug, bookmark created at`. If these are what you need, I don't want you to have to pull everything in bookmark, article, and author/profile etc.

So, please review the fields again and think of what you are excluding.